feat: Update MongoDB Atlas documentation to include AutoEmbeddings (#3661)
langchain-ai/docs: src/oss/javascript/integrations/vectorstores/mongodb_atlas.mdx

Reviewer: ```suggestion
process.env.MONGODB_ATLAS_COLLECTION_NAME = "your-atlas-collection-name";
```
Preexisting, but worth to fix I guess

---
title: "MongoDB Atlas integration"

> description: "Integrate with the MongoDB atlas vector store using LangChain JavaScript."

Reviewer: ```suggestion
description: "Integrate with the MongoDB Atlas vector store using LangChain JavaScript."
```

Create and name a cluster when prompted, then find it under `Database`. Select `Browse Collections` and create either a blank collection or one from the provided sample data.

**Note:** The cluster must be MongoDB 7.0 or higher for manual embedding mode. Automated embedding mode requires [MongoDB 8.2 or higher](https://www.mongodb.com/docs/atlas/atlas-vector-search/crud-embeddings/create-embeddings-automatic/?interface=driver&language=nodejs).

### Creating a Vector Search Index

After configuring your cluster, create a vector search index on your collection. You can do this either on [Atlas](https://www.mongodb.com/docs/atlas/atlas-vector-search/tutorials/vector-search-quick-start/), [Compass](https://www.mongodb.com/docs/compass/indexes/create-vector-search-index/), or [MongoDB Shell](https://www.mongodb.com/docs/atlas/atlas-vector-search/tutorials/vector-search-quick-start/?deployment-type=atlas&embedding=byo&interface=mongosh). The index definition depends on which embedding mode you use.

> **Manual embedding** (MongoDB 7.0+): you embed documents client-side and store the vectors in a field. Use the following definition, adjusting `numDimensions` to match your embeddings model (for example, 1024 for Cohere, 1536 for OpenAI):

Reviewer: ```suggestion
**Manual embedding** (MongoDB 7.0+): you embed documents client-side and store the vectors in a field. Use the following definition, adjusting `numDimensions` to match your embeddings model.
```
Numbers here are misleading, different models support different output dimensions, oftentimes even multiple. Claim that provider = numDimensions is oversimplifying.

}
```

**Automated embedding** (MongoDB 8.2+): MongoDB generates embeddings server-side using [Voyage AI models](https://www.mongodb.com/products/platform/ai-search-and-retrieval/models). Use the `autoEmbed` field type and specify the model:

```json
{
  "fields": [
    {
      "type": "autoEmbed",
      "modality": "text",
      "path": "text",
      "model": "voyage-4"
    }
  ]
}
```

> By default, the vector store expects an index named `default`, a text field named `text`, and an embedding field named `embedding`. Initialize the vector store with field names that match your index definition.

Reviewer: ```suggestion
By default, the vector store reads from a text field named `text` and (in manual mode) writes vectors to a field named `embedding`. Override `textKey` and `embeddingKey` to match your index.
```

Author: Accepted suggestion and added an example for improved clarity after the paragraph.

## Instantiation

Once you've set up your cluster and index, initialize your vector store. The constructor accepts two forms depending on whether you use manual or automated embedding.

<Tabs>
  <Tab title="Manual embedding">
    Pass an embeddings instance as the first argument:

    ```typescript
    import { MongoDBAtlasVectorSearch } from "@langchain/mongodb";
    import { OpenAIEmbeddings } from "@langchain/openai";
    import { MongoClient } from "mongodb";

>     const client = new MongoClient(process.env.MONGODB_ATLAS_URI || "");

Reviewer: ```suggestion
    const client = new MongoClient(process.env.MONGODB_ATLAS_URI!);
```

    const collection = client
      .db(process.env.MONGODB_ATLAS_DB_NAME)
      .collection(process.env.MONGODB_ATLAS_COLLECTION_NAME);

    const embeddings = new OpenAIEmbeddings({
      model: "text-embedding-3-small",
    });

    const vectorStore = new MongoDBAtlasVectorSearch(embeddings, {
      collection,
      indexName: "vector_index", // Defaults to "default"
      textKey: "text",           // Defaults to "text"
      embeddingKey: "embedding", // Defaults to "embedding"
    });
    ```
  </Tab>
  <Tab title="Automated embedding">
    Pass only the configuration object (no embeddings argument is needed).

    ```typescript
    import { MongoDBAtlasVectorSearch } from "@langchain/mongodb";
    import { MongoClient } from "mongodb";

>     const client = new MongoClient(process.env.MONGODB_ATLAS_URI || "");

Reviewer: ```suggestion
    const client = new MongoClient(process.env.MONGODB_ATLAS_URI!);
```

const documents = [document1, document2, document3, document4];

> await vectorStore.addDocuments(documents);

Reviewer: [critical] @RaschidJFR why did you drop `ids` from here? Examples down below still use `vectorStore.delete({ ids: ["4"] })`, for example. It becomes no-op (as ids will be auto-generated ObjectIds and `_id: { $in: chunk }` won't match).

Author: You are right. I hadn't noticed that the examples were interrelated. Restoring.
